### ISSUE TITLE
Request to add imputation Using k-NN

### ISSUE THREAD

**Issue by iteachmachines** [pycaret/pycaret#171]
This can be very useful in making predictions about the missing values by finding the k’s closest neighbours to the observation with missing data and then imputing them based on the non-missing values in the neighbourhood.

Can be much more accurate than the mean, median or most frequent imputation methods (Depending on the dataset).

**Comment by pycaret** [comment]
@iteachmachines Sure. I am tagging this as enhancement for future release.

**Comment by iteachmachines** [comment]
@pycaret  Thanks.I am willing to contribute to codebase too.CAn you tell me how to proceed?

**Comment by pycaret** [comment]
@iteachmachines Hi .. At this point I haven't created any guideline but I can give you some pointers.

- Change will be have to be done mainly in preprocess.py. Look out for a class SimpleImputer and see around that you will figure out a way to inject Knn imputer there. It's from sklearn so it shouldn't be a problem. You will have to see if it behaves differently for categorical and numeric feature.
- Once 1st step is done, second step is you have to expose the change in module files (regression.py, classification.py, clustering.py, anomaly.py). Lookout for imputation_type parameter within setup function of all modules.

You can submit a PR to dev-1.0.1 branch. Please also submit a notebook with simple test case.

Let me know if you have any questions?

**Comment by iteachmachines** [comment]
Thanks nicely explained I'll look onto to it.Looking forward to dive deeper with this project. 👍

**Comment by pycaret** [comment]
# Open for contribution

**Comment by iteachmachines** [comment]
Working on it.

**Comment by pycaret** [comment]
@iteachmachines Would you be able to submit the PR in this week along with the documentation update in `classification.py` and `regression.py`. I am planning to include this change in `2.1` release next week.

**Comment by iteachmachines** [comment]
@pycaret  Sure!

**Comment by iteachmachines** [comment]
@pycaret  I'm facing some issues.I guess someone else should work in it.

**Comment by g0lemXIV** [comment]
If I find the time, I will try to implement that by Saturday, @iteachmachines could You send me any publications/articles/blog posts about implementation?

**Comment by iteachmachines** [comment]
@g0lemXIV  Teah the steps are mentioned in upward section by @pycaret refer that.

**Comment by pycaret** [comment]
@iteachmachines 

This issue has been fixed in the `pycaret-nightly` release. Please install pycaret-nightly using `pip install pycaret-nightly`. Feel free to open a new issue, in case the problem persists. This fix will be published to PyPi as `pycaret==2.2.0` on October 28, 2020.
